src/python/pants/backend/python/tasks2/pex_build_util.py

```diff
@@ -119,6 +128,68 @@ def dump_requirements(builder, interpreter, req_libs, log, platforms=None):
       locations.add(dist.location)
 
 
+def build_python_distribution_from_target(target, workdir):
+  # Create a temporary working directory, pydistworkdir
+  pydist_workdir = os.path.join(workdir, 'pydistworkdir')
+  safe_mkdir(pydist_workdir)
+  pex_name = "%s.pex" % target.name
```

Comment:
I would expect to see this keying all writes to some path under e.g. `.pants.d/pyprep/local_dists/<target fingerprint>/...` - ideally with that key computed outside of this function and handed in as a target dir vs being handed the workdir directly. it would also probably be a good idea to include some form of no-op reuse of previously built wheels on re-keys to the same fingerprint for build performance.

in order to ensure that sources are properly declared, it should likely also copy the declared sources for the target into a temporary dir in that path before invoking the `setup.py` in it. this will ensure that any temporary files that are created as part of that land in the workdir and thus don't need manual cleanup such as what you're doing for `*.egg-info` below.

Reply:
✅